lib/ansible/module_utils/csharp/Ansible.Basic.cs

```diff
@@ -242,12 +242,24 @@ public void Debug(string message)
                 LogEvent(String.Format("[DEBUG] {0}", message));
         }
 
-        public void Deprecate(string message, string version)
+        public void Deprecate(string message, string version, string collectionName = null)
         {
+            // `collectionName` is a Ansible 2.10 parameter. We accept and ignore it,
+            // to avoid modules/plugins from 2.10 conformant collections to break with
+            // new enough versions of Ansible 2.9.
             deprecations.Add(new Dictionary<string, string>() { { "msg", message }, { "version", version } });
             LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, version));
         }
 
+        public void Deprecate(string message, DateTime date, string collectionName = null)
```

Comment:
@jborean93 could such an additional overload lead to breaking of existing modules that worked with Ansible 2.9 before?

Reply:
Hmm, from the errors (https://app.shippable.com/github/ansible/ansible/runs/167901/18/console) it looks like this won't work (though I don't really understand what's going wrong there).

Reply:
It looks like Windows 2012 gets confused if overloading and default values for parameters are combined. With four simple overloads without default values it seems to work.

One potential problem where the overloads will be problematic: if `m.Deprecate(msg, null)` is called.